test/blackbox/blackbox-kotlin.spec.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * Blackbox tests are the final line of defence, that takes different real-life example documents and generate their corresponding models in all supported languages.
+ *
+ * For those languages where it is possible, the models are compiled/transpiled to ensure there are no syntax errors in generated models.
+ *
+ */
+
+import * as path from 'path';
+import * as fs from 'fs';
+import {
+  InputMetaModel,
+  InputProcessor,
+  KotlinFileGenerator,
+  KOTLIN_DEFAULT_PRESET, JAVA_COMMON_PRESET, KOTLIN_CONSTRAINTS_PRESET,
+} from '../../src';
+import { execCommand } from './utils/Utils';
+import filesToTest from './BlackBoxTestFiles';
+
+async function generate(fileToGenerateFor: string): Promise<InputMetaModel> {
+  const inputFileContent = await fs.promises.readFile(fileToGenerateFor);
+  const processor = new InputProcessor();
+  const input = JSON.parse(String(inputFileContent));
+  return processor.process(input);
+}
+
+function deleteDirectoryIfExists(directory: string) {
+  if (fs.existsSync(directory)) {
+    fs.rmSync(directory, { recursive: true });
+  }
+}
+
```

Comment:
Because it's not that important for now, as long as this runs on all other environments then windows we are good. Do you mind creating an issue for this and then link it here? Just so we don't forget about it.

```suggestion

const isWindows = process.platform === "win32";
if (isWindows) {
  // Windows environment has a weird setup that is hard to replicate, see issue X
  return;
}
```